agent/agent.go

```diff
@@ -3700,6 +3738,27 @@ func (a *Agent) ReloadConfig() error {
 	// breaking some existing behavior.
 	newCfg.NodeID = a.config.NodeID
 
+	//if auto reload is enabled, make sure we have the right certs file watched.
+	if autoReload {
+		for _, f := range []struct {
+			oldCfg string
+			newCfg string
+		}{{a.config.KeyFile, newCfg.KeyFile}, {a.config.CertFile, newCfg.CertFile}} {
+			if f.oldCfg != f.newCfg {
+				a.FileWatcher.Remove(f.oldCfg)
+				err = a.FileWatcher.Add(f.newCfg)
```

Comment:
trivial:
I wonder if there's utility in adding a `FileWatcher.Replace(old, new)` command to cut down on the weirdness of this 2-element 2-tuple loop setup.

Reply:
I was thinking about this but it make the `FileWatcher` API weird with a replace and no add nor a remove